File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        let L=0,R=numbers.length-1
        while(L<R){
            if(numbers[L]+numbers[R]===target){
                return [L+1,R+1]
            }else if(numbers[L]+numbers[R]<target){
                L++
            }else{
                R--
            }
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap={}
        for i in range(len(nums)):
            if prevMap.get(target-nums[i])!=None:
                return [prevMap.get(target-nums[i]),i]
            else:
                prevMap[nums[i]]=i